Signal/src/ViewControllers/SelectRecipientViewController.m

```diff
@@ -0,0 +1,553 @@
+
+@property (nonatomic, readonly) OWSTableViewController *tableViewController;
+
+@property (nonatomic) NSString *callingCode;
+
+@end
+
+#pragma mark -
+
+@implementation SelectRecipientViewController
+
+- (void)loadView
+{
+    [super loadView];
+
+    self.view.backgroundColor = [UIColor whiteColor];
+    [self.navigationController.navigationBar setTranslucent:NO];
+
+    _contactsViewHelper = [ContactsViewHelper new];
+    _contactsViewHelper.delegate = self;
+
+    [self createViews];
+
+    [self populateDefaultCountryNameAndCode];
+}
+
+- (void)viewDidLoad
+{
+    OWSAssert(self.tableViewController);
+
+    [super viewDidLoad];
+}
+
+- (void)viewDidAppear:(BOOL)animated
+{
+    [super viewDidAppear:animated];
+    [self.tableViewController viewDidAppear:animated];
+
+    if ([self.delegate shouldHideContacts]) {
+        [self.phoneNumberTextField becomeFirstResponder];
+    }
+}
+
+- (void)createViews
+{
+    OWSAssert(self.delegate);
+
+    _tableViewController = [OWSTableViewController new];
+    _tableViewController.delegate = self;
+    [self.view addSubview:self.tableViewController.view];
+    [_tableViewController.view autoPinWidthToSuperview];
+    [_tableViewController.view autoPinToTopLayoutGuideOfViewController:self withInset:0];
+    [_tableViewController.view autoPinToBottomLayoutGuideOfViewController:self withInset:0];
+
+    [self updateTableContents];
+
+    [self updatePhoneNumberButtonEnabling];
+}
+
+- (UILabel *)countryCodeLabel
+{
+    UILabel *countryCodeLabel = [UILabel new];
+    countryCodeLabel.font = [UIFont ows_mediumFontWithSize:18.f];
+    countryCodeLabel.textColor = [UIColor blackColor];
+    countryCodeLabel.text
+        = NSLocalizedString(@"REGISTRATION_DEFAULT_COUNTRY_NAME", @"Label for the country code field");
+    return countryCodeLabel;
+}
+
+- (UIButton *)countryCodeButton
+{
+    if (!_countryCodeButton) {
+        _countryCodeButton = [UIButton buttonWithType:UIButtonTypeCustom];
+        _countryCodeButton.titleLabel.font = [UIFont ows_mediumFontWithSize:18.f];
+        _countryCodeButton.contentHorizontalAlignment = UIControlContentHorizontalAlignmentRight;
+        [_countryCodeButton setTitleColor:[UIColor ows_materialBlueColor] forState:UIControlStateNormal];
+        [_countryCodeButton addTarget:self
+                               action:@selector(showCountryCodeView:)
+                     forControlEvents:UIControlEventTouchUpInside];
+    }
+
+    return _countryCodeButton;
+}
+
+- (UILabel *)phoneNumberLabel
+{
+    UILabel *phoneNumberLabel = [UILabel new];
+    phoneNumberLabel.font = [UIFont ows_mediumFontWithSize:18.f];
+    phoneNumberLabel.textColor = [UIColor blackColor];
+    phoneNumberLabel.text
+        = NSLocalizedString(@"REGISTRATION_PHONENUMBER_BUTTON", @"Label for the phone number textfield");
+    return phoneNumberLabel;
+}
+
+- (UITextField *)phoneNumberTextField
+{
+    if (!_phoneNumberTextField) {
+        _phoneNumberTextField = [UITextField new];
```

Comment:
I actually did this and reverted it.  I couldn't figure out a way to make it look right.  If you pick some arbitrary width, the placeholder text & short group names look wrong. You don't want the underline to be too long or too short.  

Very happy to revisit this- I've added a card.